### DISCUSSION TITLE
[Feature Request]: make older messages available on all clients connected to the device

### DISCUSSION THREAD

**Discussion by ToeiRei** [meshtastic/firmware#4597] [Ideas]
### Platform

ESP32

### Description

Imagine, you are on your mobile on the android app. You receive a message. For convenience you connect your device to the serial port of your PC or WiFi.

It would be nice to have the latest messages available so you could easily transition between devices Another nice use-case would be checking on a device I run here (client which is mostly here to be a stable repeater) and read messages in case someone was writing me.
Right now it's pretty much impossible for me to check for messages on the web client if it's not permanently logged on or using some mqtt hack-fu.

**Comment by thebentern** [comment]
You may consider looking into the Store and Forward module. I'm not sure it solved all of your intended use cases but perhaps some of the extended message storage.

**Comment by ToeiRei** [comment]
Didn't really solve anything. Right now I have to use the MQTT to store messages for some persistence

**Comment by todd-herbert** [comment]
I imagine in essence this is another request to keep a "message history" on the node itself (and in this case, have that history accessible to whatever client is currently connected). Which would be great and something I'd love to see, but is challenging to implement.
